packages/kbn-i18n/src/react/inject_i18n_provider.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+
+import { I18nProvider } from './provider';
+
+export const injectI18nProvider = <P extends object>(WrappedComponent: React.ComponentType<P>) => {
```

Comment:
__nit:__ no need to limit `P` type here with `extends object` and plain old `function` would look a bit clearer here that `const` + arrow:
```ts
export function injectI18nProvider<P>(WrappedComponent: React.ComponentType<P>) {
```